Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgePaymentMethodsViewController.swift

```diff
@@ -144,36 +144,54 @@ final class PledgePaymentMethodsViewController: UIViewController {
 
   private func goToPaymentSheet(data: PaymentSheetSetupData) {
-    PaymentSheet.FlowController
-      .create(
-        setupIntentClientSecret: data.clientSecret,
-        configuration: data.configuration
-      ) { [weak self] result in
-        guard let strongSelf = self else { return }
+    let isSetupIntent = data.configuration.allowsDelayedPaymentMethods
+    let isPaymentIntent = !isSetupIntent
+
+    let completion: (Result<PaymentSheet.FlowController, Error>) -> Void = { [weak self] result in
+      guard let strongSelf = self else { return }
+
+      switch result {
+      case let .failure(error):
+        strongSelf.viewModel.inputs.shouldCancelPaymentSheetAppearance(state: true)
+        strongSelf.messageDisplayingDelegate?
+          .pledgeViewController(strongSelf, didErrorWith: error.localizedDescription)
+      case let .success(paymentSheetFlowController):
+        let topViewController = strongSelf.navigationController?.topViewController
+        let paymentSheetShownWithinPledgeContext = topViewController is PledgeViewController
+
+        if paymentSheetShownWithinPledgeContext {
+          strongSelf.paymentSheetFlowController = paymentSheetFlowController
+          strongSelf.paymentSheetFlowController?.presentPaymentOptions(from: strongSelf) { [weak self] in
+            guard let strongSelf = self else { return }
 
-        switch result {
-        case let .failure(error):
-          strongSelf.viewModel.inputs.shouldCancelPaymentSheetAppearance(state: true)
-          strongSelf.messageDisplayingDelegate?
-            .pledgeViewController(strongSelf, didErrorWith: error.localizedDescription)
-        case let .success(paymentSheetFlowController):
-          let topViewController = strongSelf.navigationController?.topViewController
-          let paymentSheetShownWithinPledgeContext = topViewController is PledgeViewController
-
-          if paymentSheetShownWithinPledgeContext {
-            strongSelf.paymentSheetFlowController = paymentSheetFlowController
-            strongSelf.paymentSheetFlowController?.presentPaymentOptions(from: strongSelf) { [weak self] in
-              guard let strongSelf = self else { return }
-
-              strongSelf.confirmPaymentResult(with: data.clientSecret)
-            }
+            let selectedMethod: PaymentSourceSelected = isPaymentIntent ?
+              .paymentIntentClientSecret(data.clientSecret) :
+              .setupIntentClientSecret(data.clientSecret)
+
+            strongSelf.confirmPaymentResult(with: selectedMethod)
           }
-          strongSelf.viewModel.inputs.stripePaymentSheetDidAppear()
         }
+        strongSelf.viewModel.inputs.stripePaymentSheetDidAppear()
       }
+    }
+
+    if isSetupIntent {
+      PaymentSheet.FlowController.create(
+        setupIntentClientSecret: data.clientSecret,
+        configuration: data.configuration,
+        completion: completion
+      )
+    } else {
+      PaymentSheet.FlowController
```

Comment:
@scottkicks So here's a place where I'm stuck - if I tap `New Payment Method` in the payment sheet, this code is called, but it doesn't appear to actually display the `FlowController`.